pkg/controller/handlers/knowledgesource/knowledgesource.go

```diff
@@ -64,17 +63,77 @@ func safeStatusSave(ctx context.Context, c kclient.Client, source *v1.KnowledgeS
 	// This should be the error from the last loop, which should be a conflict
 	return err
 }
+
+func reconcileFiles(ctx context.Context, c kclient.Client, existingFiles, newFiles []v1.KnowledgeFile, complete bool) error {
+	existingNames := map[string]v1.KnowledgeFile{}
+	for _, file := range existingFiles {
+		existingNames[file.Name] = file
+	}
+
+	newNames := map[string]v1.KnowledgeFile{}
+	for _, file := range newFiles {
+		newNames[file.Name] = file
+	}
+
+	for newName, newFile := range newNames {
+		existingFile, ok := existingNames[newName]
+		if !ok {
+			if err := c.Create(ctx, &newFile); apierror.IsAlreadyExists(err) {
+				if err := c.Get(ctx, kclient.ObjectKeyFromObject(&existingFile), &existingFile); err != nil {
```

Comment:
I think if there is a conflict, existingFile will be empty and this will return error to fail the sync. 

should be `kclient.ObjectKeyFromObject(&newFile)`?

Reply:
let me look

Reply:
youre right, good catch